src/modules/adaptors/Aura/AuraERC4626Adaptor.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.21;
+
+import { BaseAdaptor, ERC20, SafeTransferLib, Cellar, PriceRouter, Math } from "src/modules/adaptors/BaseAdaptor.sol";
+import { IBaseRewardPool } from "src/interfaces/external/Aura/IBaseRewardPool.sol";
+import { ERC4626 } from "@solmate/mixins/ERC4626.sol";
+import { ERC4626Adaptor } from "src/modules/adaptors/ERC4626Adaptor.sol";
+
+/**
+ * @title Aura ERC4626 Adaptor
+ * @dev This adaptor is specifically for Aura contracts.
+ * @notice Carries out typical ERC4626Adaptor functionality and allows Cellars to claim rewards from AURA pools
+ * @author crispymangoes, 0xEinCodes
+ * NOTE: Transferrance of aura-wrapped BPT is not alowed as per their contracts: ref - https://etherscan.io/address/0xdd1fe5ad401d4777ce89959b7fa587e569bf125d#code#F1#L254
+ */
+contract AuraERC4626Adaptor is ERC4626Adaptor {
+    using SafeTransferLib for ERC20;
+    using Math for uint256;
+
+    //==================== Adaptor Data Specification ====================
+    // adaptorData = abi.encode(address auraPool) // TODO: CRISPY, the adaptor data comes out as an address even if we specify a different interface such as IERC4626 or ERC4626 right? So this should stay as `address auraPool` or should it be `ERC4626 auraPool`
```

Comment:
So it does not really matter if you encode an address, interface, or even contract type(like ERC20) they are all encoded the same way. So it is totally fine to leave this as an aura pool address, then remove the TODO comment. When ever we are decoding the adaptor data, it will be casted into whatever interface we need it to be. Kinda like how if you encode a `uint8` it encodes it as if it were a `uint256` but when you decode the data, it is type casted back to a `uint8`

Reply:
Great, thank you for the clarification! I've removed the TODO comments within the `AuraERC4626Adaptor.sol` file.